lesson_19/tests/test_cart_page_suit.py

```diff
@@ -0,0 +1,67 @@
+import time
+import pytest
+
+from ..page_objects.computer_product_page import ComputerProductPage
+from ..utilities.config_parser import get_test_data
+
+
+@pytest.mark.regression
+def test_cart_page(open_computers_page, open_login_page):
+    login_page = open_login_page
+
+    email, password = get_test_data()
+    main_page = login_page.set_password(password).set_email(email).click_login_button()
+
+    computer_page = main_page.open_computer_page()
+    product_page = computer_page.add_product_to_cart(product_index=2)
+
+    description = computer_page.description
+    price = computer_page.price
+
+    title = product_page.get_product_title()
+    product_price = product_page.get_current_price()
+
+    assert title == description and price == product_price, 'Product is added properly'
+
+    product_page.set_processor_radio('fast').set_ram_radio('8_gb').set_hdd_radio('400_gb')
+    product_page.set_additional_options(['image_viewer', 'office_suite', 'other_office_suite'])
+    product_page.set_qty('10')
+
+    processor_cost = float(product_page.set_processor_radio(value='fast', get_text=True))
+    ram_cost = float(product_page.set_ram_radio(value='8_gb', get_text=True))
+    hdd_cost = float(product_page.set_hdd_radio(value='400_gb', get_text=True))
+    product_page.set_additional_options(['image_viewer', 'office_suite', 'other_office_suite'], get_text=True)
+    additional_options = sum(ComputerProductPage.prices)
+
+    product_page.add_to_cart()
+    # Needed timeout to allow ajax send product to cart
+    time.sleep(1)
```

Comment:
sleep is not a solution. You can implement your custom wait function for such situations. I will show it at next lesson

Reply:
Custom waiter is created and put to the utilities module